lib/manageiq/password.rb

```diff
@@ -102,103 +89,62 @@ def self.try_encrypt(str)
       encrypted?(str) ? str : encrypt(str)
     end
 
-    # @returns [ver, enc]
-    def self.split(encrypted_str)
-      if encrypted_str.nil? || encrypted_str.empty?
-        [nil, encrypted_str]
-      else
-        if encrypted_str =~ REGEXP_START_LINE
-          [$1, $2]
-        elsif legacy = extract_erb_encrypted_value(encrypted_str)
-          if legacy =~ REGEXP_START_LINE
-            [$1, $2]
-          else
-            ["0", legacy]
-          end
-        else
-          [nil, encrypted_str]
-        end
-      end
-    end
-
     def self.key_root
-      @@key_root ||= ENV["KEY_ROOT"]
+      @key_root ||= ENV["KEY_ROOT"]
     end
 
     def self.key_root=(key_root)
-      clear_keys
-      @@key_root = key_root
-    end
-
-    def self.clear_keys
-      @@all_keys = nil
-    end
-
-    def self.all_keys
-      keys.values
+      @key_root = key_root
+      @key = nil
     end
 
-    def self.keys
-      @@all_keys ||= {"v2" => load_v2_key}.delete_if { |_n, v| v.nil? }
-    end
-
-    def self.v2_key
-      keys["v2"]
-    end
-
-    def self.load_v2_key
-      load_key_file("v2_key") || begin
+    def self.key
+      @key ||= load_key_file("v2_key") || begin
         key_file = File.expand_path("v2_key", key_root)
-        msg = <<-EOS
-  #{key_file} doesn't exist!
-  On an appliance, it should be generated on boot by evmserverd.
+        msg = <<~EOS
```

Comment:
glad you're testing multiple ruby versions. can never remember when this was added